# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerPrivate.h
# Commit: 2fd61eb5c69c
# Full Hash: 2fd61eb5c69ce9ac806048a35c7a7a88bf4b9652
# Author: Perry Jiang <perry@mozilla.com>
# Date: 2020-04-21 09:42:20
# Regressor Bug: 1618546
# File Overlap Count: 2
# Description:
#   Bug 1618546 - give worker debugger globals their own clients r=asuth,webidl,smaug
#   
#   - Worker debugger globals gets a client with a null principal
#   - Ensure globals are created before script loads
#   - Introduce WorkerGlobalScopeBase to share code
# ==============================================================================

diff -r 9ccf9f680b64 -r 2fd61eb5c69c dom/workers/WorkerPrivate.h
--- a/dom/workers/WorkerPrivate.h	Tue Apr 21 06:09:28 2020 +0000
+++ b/dom/workers/WorkerPrivate.h	Tue Apr 21 06:50:53 2020 +0000
@@ -9,33 +9,36 @@
 #define mozilla_dom_workers_workerprivate_h__
 
 #include "MainThreadUtils.h"
-#include "mozilla/dom/WorkerCommon.h"
-#include "mozilla/dom/WorkerStatus.h"
-#include "mozilla/Assertions.h"
+#include "ScriptLoader.h"
+#include "js/ContextOptions.h"
 #include "mozilla/Attributes.h"
 #include "mozilla/CondVar.h"
 #include "mozilla/DOMEventTargetHelper.h"
 #include "mozilla/Maybe.h"
 #include "mozilla/MozPromise.h"
+#include "mozilla/PerformanceCounter.h"
 #include "mozilla/RelativeTimeline.h"
 #include "mozilla/Result.h"
 #include "mozilla/StorageAccess.h"
+#include "mozilla/ThreadBound.h"
 #include "mozilla/ThreadSafeWeakPtr.h"
+#include "mozilla/UniquePtr.h"
+#include "mozilla/UseCounter.h"
+#include "mozilla/dom/ClientSource.h"
+#include "mozilla/dom/RemoteWorkerChild.h"
+#include "mozilla/dom/Worker.h"
+#include "mozilla/dom/WorkerCommon.h"
+#include "mozilla/dom/WorkerLoadInfo.h"
+#include "mozilla/dom/WorkerScope.h"
+#include "mozilla/dom/WorkerStatus.h"
+#include "mozilla/dom/workerinternals/JSSettings.h"
+#include "mozilla/dom/workerinternals/Queue.h"
 #include "nsContentUtils.h"
 #include "nsIContentSecurityPolicy.h"
 #include "nsIEventTarget.h"
 #include "nsILoadInfo.h"
 #include "nsTObserverArray.h"
 
-#include "js/ContextOptions.h"
-#include "mozilla/dom/RemoteWorkerChild.h"
-#include "mozilla/dom/Worker.h"
-#include "mozilla/dom/WorkerLoadInfo.h"
-#include "mozilla/dom/workerinternals/JSSettings.h"
-#include "mozilla/dom/workerinternals/Queue.h"
-#include "mozilla/PerformanceCounter.h"
-#include "mozilla/ThreadBound.h"
-
 class nsIThreadInternal;
 
 namespace mozilla {
@@ -453,7 +456,7 @@
 
   void DumpCrashInformation(nsACString& aString);
 
-  bool EnsureClientSource();
+  ClientType GetClientType() const;
 
   bool EnsureCSPEventListener();
 
@@ -461,10 +464,6 @@
 
   void EnsurePerformanceCounter();
 
-  Maybe<ClientInfo> GetClientInfo() const;
-
-  const ClientState GetClientState() const;
-
   bool GetExecutionGranted() const;
   void SetExecutionGranted(bool aGranted);
 
@@ -474,10 +473,6 @@
   JSExecutionManager* GetExecutionManager() const;
   void SetExecutionManager(JSExecutionManager* aManager);
 
-  const Maybe<ServiceWorkerDescriptor> GetController();
-
-  void Control(const ServiceWorkerDescriptor& aServiceWorker);
-
   void ExecutionReady();
 
   PerformanceStorage* GetPerformanceStorage();
@@ -624,10 +619,9 @@
     return GetServiceWorkerDescriptor().Scope();
   }
 
-  nsIURI* GetBaseURI() const {
-    AssertIsOnMainThread();
-    return mLoadInfo.mBaseURI;
-  }
+  // This value should never change after the script load completes. Before
+  // then, it may only be called on the main thread.
+  nsIURI* GetBaseURI() const { return mLoadInfo.mBaseURI; }
 
   void SetBaseURI(nsIURI* aBaseURI);
 
@@ -1047,6 +1041,8 @@
 
   void ReportUseCounters();
 
+  UniquePtr<ClientSource> CreateClientSource();
+
   Maybe<nsILoadInfo::CrossOriginEmbedderPolicy> GetOwnerEmbedderPolicy() const;
 
   class EventTarget;
@@ -1070,7 +1066,7 @@
   // This is the worker name for shared workers and dedicated workers.
   nsString mWorkerName;
 
-  WorkerType mWorkerType;
+  const WorkerType mWorkerType;
 
   // The worker is owned by its thread, which is represented here.  This is set
   // in Constructor() and emptied by WorkerFinishedRunnable, and conditionally
@@ -1194,8 +1190,6 @@
 
     RefPtr<MemoryReporter> mMemoryReporter;
 
-    UniquePtr<ClientSource> mClientSource;
-
     // While running a nested event loop, whether a sync loop or a debugger
     // event loop we want to keep track of which global is running it, if any,
     // so runnables that run off that event loop can get at that information. In